com.liferay.blade.eclipse.provider/src/com/liferay/blade/eclipse/provider/JavaFileJDT.java

```diff
@@ -558,4 +562,9 @@ protected File getFile() {
 		return null;
 	}
 
+	private void resetFile() {
+		CUCache.unget(_file);
```

Comment:
We may have a caching problem related to names.  But we should not always unget the cache each time we call "findImports".  What is we loaded a java file and then called findImports 10 times in a row.  That woudl be 10 cache ungets.

We should only unget the cache when we actually need it.  Namely, the file has changed.  

Please redesign this fix to only "unget" the file when we actually need to.  Your solution will be slow, since it is "unget"ing the cache too often.